### PR TITLE
Add support for running unit tests

### PR DESCRIPTION
**Pull request by Hinton**
Tests can currently on be run for background services. Since the folders have to be defined in the `karma.conf.js` file.
It's probably easiest to hold off testing the popup app until everything is typescript.

### CODE REVIEW

File: src/services/utils.service.ts [hunk 1]
```diff
@@ -186,28 +187,18 @@ export default class UtilsService implements UtilsServiceInterface {
...
 
-                if (typeof tldjs !== 'undefined' && tldjs) {
-                    const domain = tldjs.getDomain(url.hostname);
-                    if (domain != null) {
-                        return domain;
-                    }
-                }
-
-                return url.hostname;
+                return tldjs.getDomain(url.hostname);
```
Comment (kspearrin) [inline src/services/utils.service.ts]: Is there no case when this returns null? Seems like logic might have changed here.

File: tsconfig.json [hunk 2]
```diff
@@ -4,7 +4,10 @@
     "module": "es6",
     "target": "ES2016",
     "allowJs": true,
-    "sourceMap": true
+    "sourceMap": true,
+    "types": [
+      "jasmine"
```
Comment (kspearrin) [inline tsconfig.json]: Shouldnt the types be picked up automatically?

File: src/services/utils.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        it('should handle valid urls', () => {
+            expect(UtilsService.getDomain('https://bitwarden.com')).toBe('bitwarden.com');
+            expect(UtilsService.getDomain('http://bitwarden.com')).toBe('bitwarden.com');
+            expect(UtilsService.getDomain('http://vault.bitwarden.com')).toBe('bitwarden.com');
+            expect(UtilsService.getDomain('https://user:password@bitwarden.com:8080/password/sites?and&query#hash')).toBe('bitwarden.com');
+        });
+
+        it('should support localhost and IP', () => {
+            expect(UtilsService.getDomain('https://localhost')).toBe('localhost');
```
Comment (kspearrin) [inline src/services/utils.service.spec.ts]: What about a `https://bitwarden` test?

File: src/services/utils.service.ts (hunk 1 above)
Comment (Hinton) [reply src/services/utils.service.ts]: Yea, for the specified test cases the code was never triggered, so cleaned it up to simplify.

File: tsconfig.json (hunk 2 above)
Comment (Hinton) [reply tsconfig.json]: If you use import they get found automatically, but for the tests file it's useful to not have to write `import { describe, it ..... } from 'jasmine'` in every single file.

**Pull request merged**